src/librustc_typeck/check/mod.rs

```diff
@@ -3761,17 +3761,64 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                  error_code: &str,
                                  c_variadic: bool,
                                  sugg_unit: bool| {
+            let (span, arg_spans) = match &expr.kind {
+                hir::ExprKind::Call(hir::Expr { span, .. }, args) => (
```

Comment:
Besides the number of elements to skip and one span, the logic is common between `ExprKind::Call` and `ExprKind::MethodCall` and can be moved to outside of the match.

Reply:
done